homeassistant/components/devolo_home_control/switch.py

```diff
@@ -0,0 +1,161 @@
+"""Platform for light integration."""
+import logging
+
+from devolo_home_control_api.homecontrol import get_sub_device_uid_from_element_uid
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import DOMAIN
+
+logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Get all devices and setup the switch devices via config entry."""
+    devices = hass.data[DOMAIN]["homecontrol"].binary_switch_devices
+
+    devices_list = []
+    for device in devices:
+        for i in range(len(device.binary_switch_property)):
+            devices_list.append(
+                DevoloSwitch(
+                    hass=hass,
+                    device_instance=device,
+                    sub_uid=get_sub_device_uid_from_element_uid(
+                        [*device.binary_switch_property][i]
+                    ),
+                )
+            )
+    async_add_entities(devices_list, False)
+
+
+class DevoloSwitch(SwitchDevice):
+    """Representation of an Awesome Light."""
+
+    def __init__(self, hass, device_instance, sub_uid):
+        """Initialize an devolo Switch."""
+        self._device_instance = device_instance
+
+        # Create the unique ID
+        if sub_uid is not None:
+            self._unique_id = self._device_instance.uid + "#" + str(sub_uid)
```

Comment:
Please us f-strings instead of string concatenation.

Reply:
If I remember correctly someone said to me, not to use f-strings. Or was that just meant within logger messages?

Reply:
Yeah, for logging messages we should use the old style formatting operator. Otherwise we use f-strings as far as possible.

Reply:
Remove this section